### PR TITLE
Add Ruby Tests

### PR DESCRIPTION
**Pull request by jamesnakagawa**
Mirrors the Python tests, with a couple of changes:

- Ruby version of this library does not return the score
- Ruby version of this library does not provide for the parameter `use_terminal_gap_penalty`, which is always 0 in this version
- For insertions and deletions, gaps are placed at the beginning of the aligned string for some reason.

An example of the last point above:

```
reference: 
TGGAAGGGCTAATTCACTCCCAACGAAGACAAGATATCCTTGATCTGTGGATCTACCACACA
deletion:         ^^^^^^

sequence:
TGGAAGGGCTAATTCACTGAAGACAAGATATCCTTGATCTGTGGATCTACCACACA

in python:
TGGAAGGGCTAATTCACT------GAAGACAAGATATCCTTGATCTGTGGATCTACCACACA

in ruby:
------TGGAAGGGCTAATTCACTGAAGACAAGATATCCTTGATCTGTGGATCTACCACACA
```

### CODE REVIEW

File: test_alignment.rb [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  #   expected_score2 = 6*7 - 11
+  #
+  #   aref1, aseq1, score1 = align_it_aa_py(ref0, seq1, *DEFAULT_PENALTIES)
+  #   aref2, aseq2, score2 = align_it_aa_py(ref0, seq2, *DEFAULT_PENALTIES)
+  #
+  #   assert_equal [aref1, aseq1], [ref0, seq1]
+  #   assert_equal [aref2, aseq2], [ref0, seq2]
+  #   assert_equal score1, expected_score1
+  #   assert_equal score2, expected_score2
+  # end
```
Comment (nathanielknight) [inline test_alignment.rb]: If these tests are useful, we should add them. If they're not, they should be deleted.

**Review by nathanielknight — CHANGES_REQUESTED** [review]
I would prefer if we didn't check-in any commented-out code.

File: test_alignment.rb [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        '------TGGAAGGGCTAATTCACTGAAGACAAGATATCCTTGATCTGTGGATCTACCACACA'
+    gap_open_penalty = 15
+    gap_extend_penalty = 5
+
+    aligned_ref, aligned_seq = align_it(ref, seq, gap_open_penalty, gap_extend_penalty)
+    assert_equal ref, aligned_ref
+    assert_equal expected_aseq, aligned_seq
+  end
+
+  def test_align_it_aa_match
```
Comment (donkirkby) [inline test_alignment.rb]: This test doesn't mean anything without the score. It was just showing that the Python amino acid alignment used a variable scoring matrix.

File: test_alignment.rb [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  def test_align_it_mismatch
+    ref = 'TGGAAGGGCTAATTCACTCCCAACGAAGACAAGATATCCTTGATCTGTGGATCTACCACACACAA'
+    seq = 'TGGAAGGGATAATTCACTCCCAACGAAGACAAGATATCCTTGATCTGTGGATCTACCACACACAA'
+    # mismatch:    ^
+    aligned_ref, aligned_seq = align_it(ref, seq, *DEFAULT_PENALTIES)
+    assert_equal ref, aligned_ref
+    assert_equal seq, aligned_seq
+  end
+
+  def test_align_it_insert
```
Comment (donkirkby) [inline test_alignment.rb:28]: It looks like the Ruby wrapper for `align_it` sets up the scoring matrix incorrectly. (Matches and mismatches score the same.) Make this test fail, by using the same expectation as the Python code, and then mark it as ignored with a link to the issue. (I'll write up an issue.) We'll have to check whether `align_it` actually gets called by Recall or the upload scripts. (Hopefully not.) Then we can fix it or remove it.

File: test_alignment.rb [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    seq = 'RRRDRRR'
+    # Main difference is that align_it_aa_rb uses same score for all combos.
+
+    aref, aseq = align_it_aa(ref, seq, *DEFAULT_PENALTIES)
+
+    assert_equal ref, aref
+    assert_equal seq, aseq
+  end
+
+  #########################################
```
Comment (donkirkby) [inline test_alignment.rb]: Do we need the Python flavour from Ruby?

**Review by donkirkby — CHANGES_REQUESTED** [review]

File: test_alignment.rb (hunk 3 above)
Comment (jamesnakagawa) [reply test_alignment.rb:28]: Oh, it definitely definitely does. Upload scripts for sure - maybe Recall, too.

Comment (donkirkby) [reply test_alignment.rb:28]: OK, I was wrong. The scores are supposed to both be positive. It's just that the Python scores are higher than the Ruby ones, so you need to scale down your gap penalties. Check what penalties we call with in Recall or the upload scripts, but I would guess 3 and 1 for similar behaviour to the Python wrapper.

File: test_alignment.rb (hunk 1 above)
Comment (jamesnakagawa) [reply test_alignment.rb]: Ok, I'll delete.

**Review by nathanielknight — APPROVED** [review]

**Review by donkirkby — APPROVED** [review]

**Pull request merged**